Mathlib/FieldTheory/LinearlyDisjoint.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.FieldTheory.Adjoin
+import Mathlib.LinearAlgebra.TensorProductBasis
+
+/-!
+
+# Linearly disjoint
+
+This file contains basics about the linearly disjoint of fields.
+
+## Mathematical background
+
+(<https://en.wikipedia.org/wiki/Linearly_disjoint>) In mathematics, algebras `A`, `B` over a field
+`F` inside some field extension `E` of `F` are said to be linearly disjoint over `F` if the
+following equivalent conditions are met:
+
+- The map `A ⊗[F] B → A ⊔ B`, `x ⊗ₜ[F] y ↦ x * y` is injective.
+
+- Any `F`-basis of `A` remains linearly independent over `B`.
+
+- If `{ u_i }`, `{ v_j }` are `F`-bases for `A`, `B`, then the products `{ u_i * v_j }` are
+  linearly independent over `F`.
+
+Our definition `IntermediateField.LinearlyDisjoint` is very closed to the second equivalent
+definition in the above.
+
+(<https://mathoverflow.net/questions/8324>) For two abstract fields `E` and `K` over `F`, they are
+called linearly disjoint over `F` (`LinearlyDisjoint F E K`), if `E ⊗[F] K` is a field.
+In this case, it can be shown that at least one of `E / F` and `K / F` are algebraic, and if this
+holds, then it is equivalent to the above `IntermediateField.LinearlyDisjoint`.
+
+The advantage of `LinearlyDisjoint` is that it is preserved under algebra isomorphisms, while for
+`IntermediateField.LinearlyDisjoint` this is not so easy to prove, in fact it's wrong if both of the
+extensions are not algebraic.
+
+## Main definitions
+
+- ...
+
+## Main results
+
+- ...
+
+## Tags
+
+linearly disjoint, linearly independent, tensor product
+
+## TODO
+
+- ...
+
+-/
+
+open scoped Classical Polynomial TensorProduct
+
+open FiniteDimensional Polynomial IntermediateField Field
+
+noncomputable section
+
+universe u v w
+
+variable (F : Type u) (E : Type v) [Field F] [Field E] [Algebra F E]
+
+variable (K : Type w) [Field K] [Algebra F K]
+
+namespace IntermediateField
+
+variable {F E}
+
+variable (A B : IntermediateField F E)
+
+/-- Two intermediate fields `A` and `B` of `E / F` are called linearly disjoint, if any `F`-linearly
+independent subset of `A` remains linearly independent over `B`. Marked as `protected` because later
+we will define linearly disjoint for two abstract fields over a base field. -/
+protected def LinearlyDisjoint := ∀ {a : Set A}, LinearIndependent F (fun x : a ↦ x.1) →
```

Comment:
If we follow the precedent `LinearIndependent` and `[AlgebraicIndependent](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/AlgebraicIndependent.html#AlgebraicIndependent)` this would be called `LinearDisjoint` ...

Reply:
Should I also change the file name to `LinearDisjoint.lean`?

Reply:
Probably; the file that defines LinearIndependent is called Mathlib.LinearAlgebra.LinearIndependent.